src/Control/Email/Email.php

```diff
@@ -2,236 +2,181 @@
      *    This code is licensed under a Creative Commons Attribution-ShareAlike 2.5 License
      *    http://creativecommons.org/licenses/by-sa/2.5/
      */
-    public static function is_valid_address($address)
+    public static function is_valid_address(string $address): bool
     {
-        $validator = new EmailValidator();
-        return $validator->isValid($address, new RFCValidation());
+        $validator = Injector::inst()->get(EmailValidator::class);
+        $rfcValidation = Injector::inst()->get(RFCValidation::class);
+        return $validator->isValid($address, $rfcValidation);
     }
 
-    /**
-     * Get send_all_emails_to
-     *
-     * @return array Keys are addresses, values are names
-     */
-    public static function getSendAllEmailsTo()
+    public static function getSendAllEmailsTo(): array
     {
-        return static::mergeConfiguredEmails('send_all_emails_to', 'SS_SEND_ALL_EMAILS_TO');
+        return static::mergeConfiguredAddreses('send_all_emails_to', 'SS_SEND_ALL_EMAILS_TO');
     }
 
-    /**
-     * Get cc_all_emails_to
-     *
-     * @return array
-     */
-    public static function getCCAllEmailsTo()
+    public static function getCCAllEmailsTo(): array
     {
-        return static::mergeConfiguredEmails('cc_all_emails_to', 'SS_CC_ALL_EMAILS_TO');
+        return static::mergeConfiguredAddreses('cc_all_emails_to', 'SS_CC_ALL_EMAILS_TO');
     }
 
-    /**
-     * Get bcc_all_emails_to
-     *
-     * @return array
-     */
-    public static function getBCCAllEmailsTo()
+    public static function getBCCAllEmailsTo(): array
     {
-        return static::mergeConfiguredEmails('bcc_all_emails_to', 'SS_BCC_ALL_EMAILS_TO');
+        return static::mergeConfiguredAddreses('bcc_all_emails_to', 'SS_BCC_ALL_EMAILS_TO');
     }
 
-    /**
-     * Get send_all_emails_from
-     *
-     * @return array
-     */
-    public static function getSendAllEmailsFrom()
+    public static function getSendAllEmailsFrom(): array
     {
-        return static::mergeConfiguredEmails('send_all_emails_from', 'SS_SEND_ALL_EMAILS_FROM');
+        return static::mergeConfiguredAddreses('send_all_emails_from', 'SS_SEND_ALL_EMAILS_FROM');
     }
 
     /**
      * Normalise email list from config merged with env vars
      *
-     * @param string $config Config key
-     * @param string $env Env variable key
-     * @return array Array of email addresses
+     * @return Address[]
      */
-    protected static function mergeConfiguredEmails($config, $env)
-    {
-        // Normalise config list
-        $normalised = [];
-        $source = (array)static::config()->get($config);
-        foreach ($source as $address => $name) {
-            if ($address && !is_numeric($address)) {
-                $normalised[$address] = $name;
-            } elseif ($name) {
-                $normalised[$name] = null;
-            }
+    private static function mergeConfiguredAddreses(string $configKey, string $envKey): array
```

Comment:
Why are we renaming this? It doesn't seem necessary to achieve this PR's purpose?
If you do insist on making this change, please change it to `mergeConfiguredAddresses`

Reply:
Renamed it to `mergeConfiguredAddresses`

Reply:
Can you please answer the original question? Why are we renaming this? It's additional upgrade pain for, as far as I can tell, a purely aesthetic reason? Is there any _real_ benefit to this to justify the upgrade pain in projects?

Reply:
I changed it from protected to private so it's removed API (it says removed in changelog). The name doesn't really matter.  It's not something used in projects, it's just a shared method for `getSendAllEmailsTo()`, `getCCAllEmailsTo()` and a couple of other methods

Reply:
Ahh, I hadn't noticed it was changed to private. Fair enough then.